Reqnroll.VisualStudio/Editor/Commands/FindUnusedStepDefinitionsCommand.cs

```diff
@@ -0,0 +1,206 @@
+        }
+
+
+        public override bool PreExec(IWpfTextView textView, DeveroomEditorCommandTargetKey commandKey, IntPtr inArgs = default)
+        {
+            Logger.LogVerbose("Find Unused Step Definitions");
+
+            var textBuffer = textView.TextBuffer;
+            var fileName = GetEditorDocumentPath(textBuffer);
+            var triggerPoint = textView.Caret.Position.BufferPosition;
+
+            var project = IdeScope.GetProject(textBuffer);
+            if (project == null || !project.GetProjectSettings().IsReqnrollProject)
+            {
+                IdeScope.Actions.ShowProblem(
+                    "Unable to find step definition usages: the project is not detected to be a Reqnroll project or it is not initialized yet.");
+                return true;
+            }
+
+            var reqnrollTestProjects = IdeScope.GetProjectsWithFeatureFiles()
+                .Where(p => p.GetProjectSettings().IsReqnrollTestProject)
+                .ToArray();
+
+            if (reqnrollTestProjects.Length == 0)
+            {
+                IdeScope.Actions.ShowProblem(
+                    "Unable to find step definition usages: could not find any Reqnroll project with feature files.");
+                return true;
+            }
+
+            var asyncContextMenu = IdeScope.Actions.ShowAsyncContextMenu(PopupHeader);
+            Task.Run(
+                () => FindUnusedStepDefinitionsInProjectsAsync(reqnrollTestProjects, fileName, triggerPoint, asyncContextMenu,
+                    asyncContextMenu.CancellationToken), asyncContextMenu.CancellationToken);
+            return true;
+        }
+
+        private async Task FindUnusedStepDefinitionsInProjectsAsync(IProjectScope[] reqnrollTestProjects, string fileName,
+            SnapshotPoint triggerPoint, IAsyncContextMenu asyncContextMenu, CancellationToken cancellationToken)
+        {
+            var summary = new UnusedStepDefinitionSummary();
+
+            try
+            {
+                await FindUsagesInternalAsync(reqnrollTestProjects, fileName, triggerPoint, asyncContextMenu,
+                    cancellationToken, summary);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogException(MonitoringService, ex);
+                summary.WasError = true;
+            }
+
+            if (summary.WasError)
+                asyncContextMenu.AddItems(new ContextMenuItem("Could not complete find operation because of an error"));
+            else if (summary.UnusedStepDefinitions == 0)
+                asyncContextMenu.AddItems(
+                    new ContextMenuItem("Could not find any unused step definitions at the current position"));
+            //TODO: determine if this required/needed
+            //else if (summary.UsagesFound == 0) asyncContextMenu.AddItems(new ContextMenuItem("Could not find any usage"));
+
+            //TODO: modify monitoring to include finding unused step definitions
```

Comment:
I would definitely include the monitoring, so that we can see how users will use this feature. You can make a new telemetry event similar to the find usages.

Reply:
Done. Added MonitorCommandFindUnusedSStepDefinitions to IMonitoringService and its two implementations (MonitoringService and NullVsIdeScope.cs)